File: desktopclient/src/main/java/com/goexp/galgame/gui/util/FXMLLoaderProxy.java
package com.goexp.galgame.gui.util;

import javafx.fxml.FXMLLoader;

import java.io.IOException;

public class FXMLLoaderProxy<N, C> {

    public final N node;
    public final C controller;
    private final FXMLLoader loader;

    public FXMLLoaderProxy(final String path) {
        this.loader = new FXMLLoader(getClass().getClassLoader().getResource(path));

        node = load();
        controller = loader.getController();
    }

    private N load() {
        try {
            return loader.load();
        } catch (IOException e) {
            e.printStackTrace();
            throw new RuntimeException(e);
        }
    }
}

File: desktopclient/src/main/scala/com/goexp/galgame/gui/util/FXMLLoaderProxy.scala
package com.goexp.galgame.gui.util

import java.net.URL

import javafx.fxml.FXMLLoader

class FXMLLoaderProxy[N, C](val url: URL) {
  final private val loader = new FXMLLoader(url);
  final val node: N = loader.load
  final val controller: C = loader.getController

  def this(path: String) {
    this(classOf[FXMLLoaderProxy[N, C]].getClassLoader.getResource(path))
  }

}